syntax = "proto3";

package poker;

option cc_generic_services = true;

enum Round {
  FLOP = 0;
  TURN = 1;
  RIVER = 2;
}

message InitialData {
  repeated PlayerStateData players_state = 1;

  PlayerData dealer = 2;

  float sb_amount = 3;
  float bb_amount = 4;
}

message Blinds {
  float small_blind = 1;
  float big_blind = 2;
}

message BlindsRequest {
  uint32 num_hands = 1;
}

message PlayerData {
  uint32 position = 1;
  //main or opponent

  enum PlayerType {
    MAIN = 0;
    OPPONENT = 1;
  }

  PlayerType player_type = 2;
  bool is_active = 3;
}

message PlayerCardsRequest {
  bool new_hand = 1;
  bool showdown = 2;
  PlayerData player = 3;
}

message Empty {

}

message PlayerActionRequest {
  uint32 round = 1;
  PlayerData player_data = 2;
  bool new_hand = 3;
}

message BoardCardsRequest {
  Round round = 1;
}

message PlayerStateData {
  float chips = 1;
  PlayerData player = 2;
}

message CardsData {
  repeated CardData cards = 1;
}

message CardData {
  string rank = 1;
  string suit = 2;
}

message ActionData{
  enum ActionType {
    BET = 0;
    CALL = 1;
    RAISE = 2;
    FOLD = 3;
    ALL_IN = 4;
    CHECK = 5;
  }

  ActionType action_type = 1;
  float multiplier = 2;
  float amount = 3;
}

message Amount {
  float value = 1;
}